pkg/monitoring/poller.go

```diff
@@ -0,0 +1,70 @@
+package monitoring
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/gagliardetto/solana-go"
+)
+
+type Poller interface {
+	Start(context.Context)
+	Updates() <-chan interface{}
+}
+
+func NewPoller(
+	account solana.PublicKey,
+	reader AccountReader,
+	fetchInterval time.Duration,
+	bufferCapacity uint32,
+) Poller {
+	return &solanaPollerImpl{
+		account,
+		reader,
+		fetchInterval,
+		bufferCapacity,
+		make(chan interface{}, bufferCapacity),
+	}
+}
+
+type solanaPollerImpl struct {
+	account        solana.PublicKey
+	reader         AccountReader
+	fetchInterval  time.Duration
+	bufferCapacity uint32
+	updates        chan interface{}
+}
+
+// Start should be executed as a goroutine
+func (s *solanaPollerImpl) Start(ctx context.Context) {
+	// Fetch initial data
+	data, err := s.reader.Read(ctx, s.account)
+	if err != nil {
+		log.Printf("failed initial read of the account contents for address '%s': %v", s.account.String(), err)
```

Comment:
Please use the relay `Logger` interface instead directly `log.Printf`

Reply:
Updated